spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigMapPropertySource.java

```diff
@@ -38,7 +39,7 @@
 
 	public KubernetesClientConfigMapPropertySource(CoreV1Api coreV1Api, String name, String namespace,
 			Environment environment) {
-		super(getName(name, namespace), asObjectMap(getData(coreV1Api, name, namespace, environment)));
```

Comment:
I've removed the `asObjectMap` call. It's implementation looked like this:

```
protected static Map<String, Object> asObjectMap(Map<String, Object> source) {
		return source.entrySet().stream()
				.collect(Collectors.toMap(Entry::getKey, Entry::getValue, throwingMerger(), LinkedHashMap::new));
}
```

It took a `Map` and created a `LinkedHashMap` out of it. So you take a `Map` interface, without any defined order and transform it to a `LinkedHashMap`? This is most probably an artifact of some previous implementation, in the current context - it would not make sense. 

The fact that it is `protected` also means - we can drop it; unless you declare the same package where it was declared initially, which no one should have done that anyway. imho, removing it - is safe.